src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.cs

```diff
@@ -14,6 +14,9 @@ namespace System.Text.Json.Nodes
     /// declared as an <see cref="object"/> should be deserialized as a <see cref="JsonNode"/>.
     public abstract partial class JsonNode
     {
+        // Default options instance used when calling built-in JsonNode converters.
+        private protected static readonly JsonSerializerOptions s_defaultOptions = new();
```

Comment:
Why can't this use JsonSerializerOptions.Default?

Reply:
It would result in linker warnings. This is just a dummy parameter being passed to the built-in converters used by the primitive converters if no options is provided by the user. FWIW this is just moving the dummy parameter from the generic `JsonValue<T>` to the base non-generic type.

Reply:
> It would result in linker warnings.

Why?

Reply:
`JsonSerializerOptions.Default` is marked RequiredUnreferencedCode/RequiresDynamicCode.

Reply:
Shouldn't we be concerned that we exposed `JsonSerializerOptions.Default` so that developers don't need to do `private protected static readonly JsonSerializerOptions s_defaultOptions = new();` and now we're being forced to do it anyway?

Reply:
Should we be obsoleting JsonSerializerOptions.Default and JsonSerializerOptions.Web? Seems like we have these prominent properties that we actually don't want folks using? Why is it ok as well in cases like:
https://github.com/dotnet/aspnetcore/blob/0da8ea72b5434cbe8e1207d802f2270ca2f8ad4c/src/Caching/Hybrid/src/Internal/DefaultJsonSerializerFactory.cs#L35
that these warnings are suppressed... is that type not intended to be use with NativeAOT?

Reply:
Application authors not targeting Native AOT should keep using `Default` and `Web`. Library authors who are looking to support native AOT shouldn't be hardcoding defaults in the first place. This particular use case is an outlier that reflects how `JsonNode` chooses to represent primitives.

Reply:
I still don't understand, though... how is this custom s_defaultOptions any better for NativeAOT than JsonSerializerOptions.Default?

Reply:
It's just a placeholder value that doesn't specify any resolvers and is used as a parameter in the built-in converters encapsulated by `JsonValue`:

https://github.com/dotnet/runtime/blob/4a7fe654d798a372f5786f026006437444f14f1e/src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueOfTPrimitive.cs#L34-L43

On the other hand, `JsonSerializerOptions.Default` is explicitly configured with the default reflection-based resolver.

Reply:
A `new JsonSerializerOptions()` is different than `JsonSerializerOptions.Default`, until you pass it as a parameter into one of the `JsonSerializer` methods, at which point it is automatically populated with the reflection-based resolver unless otherwise configured. It's behavior that got shipped with STJ from day one, and one of the biggest source of headaches for me :-)